Automation/S3/delete_s3_buckets.py

```diff
@@ -0,0 +1,194 @@
+parser.add_argument('--RegionName', dest='region_name', default='us-west-2', help='Region Name (default: us-west-2)')
+
+args = parser.parse_args()
+
+def main(aws_access_key_id, aws_secret_access_key, region_name):
+    '''
+    The main function for the S3 bucket deletion script
+    '''
+
+    print(Fore.GREEN + "Starting S3 bucket deletion script...")
+    try:
+        client_config = Config(
+            region_name = region_name
+        )
+
+        s3_client = boto3.client(
+            's3',
+            aws_access_key_id=aws_access_key_id,
+            aws_secret_access_key=aws_secret_access_key,
+            config=client_config
+        )
+
+        print("Retrieving S3 bucket list...")
+        list_buckets_response = s3_client.list_buckets()
+        print(Fore.GREEN + "Bucket list retrieved! Cycling through list...\n")
+
+        buckets_to_delete = []
+        for bucket in list_buckets_response['Buckets']:
+            bucket_name = bucket['Name']
+            
+            # Retrieve Bucket Information
+            print(f'Retrieving bucket: {bucket_name} information...')
+            bucket_owner_email = get_owner_email(s3_client, bucket_name)
+            bucket_object_list = get_bucket_object_list(s3_client, bucket_name)
+
+            # Display that information
+            print(Fore.GREEN + f'{"Bucket:":<{COL_SIZE}}{bucket_name}')
+            print(Fore.GREEN + f'{"OwnerEmail:":<{COL_SIZE}}{bucket_owner_email}')
+            bucket_object_list_prepared_for_deletion = []
+            if len(bucket_object_list) != 0:
+                print(Fore.GREEN + "Bucket Objects:")
+                for bucket_object in bucket_object_list:
+                    # Prepare just in case we want to delete the bucket
+                    bucket_object_list_prepared_for_deletion.append({
+                        'Key': bucket_object
+                    })
+                    print(bucket_object)
+            
+            # Prompt user for response
+            delete_bucket_response = ""
+            while delete_bucket_response == "":
+                deletion_prompt_response = input(Fore.YELLOW + "Delete bucket: (y/N/exit)")
+                if deletion_prompt_response == "y":
+                    delete_bucket_response = "True"
+                elif deletion_prompt_response == "N":
+                    delete_bucket_response = "False"
+                elif deletion_prompt_response == "exit":
+                    delete_bucket_response = "Exit"
+
+            
+            if delete_bucket_response == "True":
+                deletion_information = {
+                    'BucketName': bucket_name,
+                    'Objects': bucket_object_list_prepared_for_deletion
+                }
+                buckets_to_delete.append(deletion_information)
+                print(Fore.GREEN + "Object scheduled for deletion!")
+            elif delete_bucket_response == "Exit":
```

Comment:
What if I type `blah`?

Reply:
There's an input loop above, it will keep asking until you enter one of the defined options. I should print something out if they get it wrong!